### PR TITLE
Pr/42

### PR DESCRIPTION
**Pull request by marton-balazs-kovacs**
I made some small changes to the google url read feature that @jcolomb created.
Now it uses to upload buttons for the two sources separately. I think it worth doing so as you can change sources in one session without breaking the app.

### CODE REVIEW

File: R/mod_read_spreadsheet.R [hunk 1]
```diff
@@ -33,35 +59,75 @@ mod_read_spreadsheet_ui <- function(id){
...
-    table_data_clean <- eventReactive(valid_infosheet, {
+    table_data_clean <- eventReactive(activate(), {
+      if (valid_infosheet() == TRUE) {
       table_data() %>%
       tibble::as_tibble() %>%
       dplyr::filter_at(
         dplyr::vars(Firstname, `Middle name`, Surname),
         dplyr::any_vars(!is.na(.)))
+      } else {
+        NULL
```
Comment (jcolomb) [inline R/mod_read_spreadsheet.R:122]: maybe we should return something here, so one can see the sheet and why it was not validated ?

```suggestion
        table_data() %>%
        tibble::as_tibble() %>%
        tibble::add_row("Firstname"="warning: this infosheet did not pass validation, please see the errors that appened during validation, .before = 0)
```

Comment (marton-balazs-kovacs) [reply R/mod_read_spreadsheet.R:122]: The validation code is still running and upon loading an infosheet the modal pop up with the result of the validation that explains why it was not validated.

The reason why we do not show the sheet until it throws an error is that there is a custom code that highlights the row with missing credit info which needs the credit columns to be present in the infosheet currently.

We can change that behavior but I think that should be in a different pull request not in this one.

**Comment by marton-balazs-kovacs** [comment]
Maybe they check out this branch and run the app locally so they can see the changes. I would not publish the app either without checking.

I think using two separate buttons is pretty straightforward and it saves space from the UI. I even added tooltips to the buttons. I can highlight the input field or the button of the currently loaded field if you think that is necessary. But as you said earlier I do not think that people will use different infosheets, especially from different sources frequently. But at the same time, the app should be robust and deal with this special case, that's why I added the two buttons.

File: R/mod_read_spreadsheet.R [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #'
 #' @keywords internal
 #' @export 
-#' @importFrom shiny NS tagList 
+#' @importFrom shiny NS tagList googlesheets4
```
Comment (marton-balazs-kovacs) [inline R/mod_read_spreadsheet.R:15]: Why is this needed?

**Comment by jcolomb** [comment]
There are still unresolved comments.

File: R/mod_read_spreadsheet.R (hunk 2 above)
Comment (jcolomb) [reply R/mod_read_spreadsheet.R]: I think it is for documentation, to tell what dependencies the function has

File: R/mod_read_spreadsheet.R (hunk 1 above)
Comment (jcolomb) [reply R/mod_read_spreadsheet.R:122]: Then please create a new PR with this code, otherwise we will forget about this.

Comment (jcolomb) [reply R/mod_read_spreadsheet.R:122]: new issue created #51

File: R/mod_read_spreadsheet.R (hunk 2 above)
Comment (jcolomb) [reply R/mod_read_spreadsheet.R]: please accept or reject before we merge.

**Review by jcolomb — CHANGES_REQUESTED** [review]

**Comment by jcolomb** [comment]
All good then !
